app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -106,6 +106,7 @@ public void onReceive(Context context, Intent intent) {
     private final BroadcastReceiver showUpdateDialog = new BroadcastReceiver() {
         @Override
         public void onReceive(Context context, Intent intent) {
+            assert BuildConfig.isInternetAvailable;
```

Comment:
what's the value of this assertion?

I hadn't taken note of `BuildConfig.isInternetAvailable` before. In an ideal world that doesn't need to exist. Any code that does internety things would live in the `online` flavour and there would be a corresponding interface that does nothing in the `offline` flavour, like what you did with `AppUpdateUtil`